src/ripple/app/misc/impl/AMMUtils.cpp

```diff
@@ -201,6 +201,9 @@ deleteAMMTrustLines(
         [&](LedgerEntryType nodeType,
             uint256 const&,
             std::shared_ptr<SLE>& sleItem) -> TER {
+            // Skip AMM
+            if (nodeType == LedgerEntryType::ltAMM)
+                return tecOBJECT_NOT_FOUND;
```

Comment:
I'm not a huge fan of using `tecOBJECT_NOT_FOUND`. I think it would make more sense to find (or even create) a `ter` code that serves this purpose. Even though it'll never be exposed externally, even internally, I'd rather see something that tells me this object will be retried later. Maybe `terNO_AMM`? Or even just the base `terRETRY` (it has the advantage of not being used anywhere else).

Reply:
I actually had `terRETRY` at first. Thanks, will change.

Reply:
You're abusing ter codes to mean things they don't mean. `tecOBJECT_NOT_FOUND` is no good (the object is found), and `terRETRY` doesn't mean "skip this entry". I don't think we can use a ter code to mean "skip this entry". Consider changing the callback function to return both the ter and an enum that says to skip or not. There are other interface changes we can consider too, but I abusing ter codes doesn't seem like the right approach here.